tests/integration/base_object_test.go

```diff
@@ -56,13 +56,22 @@ func runObjectE2ETest(helper *clients.TestClient, k8sh *utils.K8sHelper, s suite
 	logger.Infof("Step 1 : Create Object Store User")
 	cosuErr := helper.ObjectUserClient.Create(namespace, userid, userdisplayname, storeName)
 	require.Nil(s.T(), cosuErr)
-	logger.Infof("Waiting 5 seconds to ensure user was created")
-	time.Sleep(5 * time.Second)
-	require.True(s.T(), helper.ObjectUserClient.UserSecretExists(namespace, userid, storeName), "make sure user secret was created")
+	logger.Infof("Waiting 10 seconds to ensure user was created")
+	time.Sleep(10 * time.Second)
 	userInfo, gosuErr := helper.ObjectUserClient.GetUser(namespace, storeName, userid)
 	require.Nil(s.T(), gosuErr)
 	require.Equal(s.T(), userid, userInfo.UserID)
 	require.Equal(s.T(), userdisplayname, *userInfo.DisplayName)
+	logger.Infof("Checking to see if the user secret has been created")
+	i := 0
+	for i = 0; i < 4 && helper.ObjectUserClient.UserSecretExists(namespace, storeName, userid) != true; i++ {
```

Comment:
since we have this loop, do we still need the 10s sleep above?

Reply:
Can we add an `assert` that verifies the user secret exists? If we only write to the log, an error won't be noticed in the future.

Reply:
The helper.ObejctUserClient.GetUser() check occasionally failed when my local CI ran with a 5 second delay.

Reply:
After all the retries, we expect it to succeed, right? We wouldn't need to assert after the first try, only if all the retries fail.

Reply:
The secret will not exist inside the loop. We should expect the secret to exist after the for loop is finished. I could replace the "if i < 4" with an assert to check if the key was successfully created or if the system failed somehow.

Reply:
Agreed, an assert after the loop sounds good